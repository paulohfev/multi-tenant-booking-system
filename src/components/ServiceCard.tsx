import { Service } from "@/types/Service";

export type ServiceCardProps = {
  service: Service;
}

const ServiceCard: React.FC<ServiceCardProps> = ({ service }) => {
  return (
    <div className="bg-white rounded-lg shadow-md p-6 hover:shadow-lg transition-shadow duration-200">
      <h3 className="text-xl font-semibold text-gray-900 mb-2">
        {service.name}
      </h3>
      <p className="text-gray-600 mb-4">{service.description}</p>
      <div className="flex items-center justify-between text-sm text-gray-500 mb-4">
        <span>Duration: {service.duration} min</span>
        <span className="font-semibold text-gray-900">
          ${service.price}
        </span>
      </div>
      <button className="w-full bg-blue-600 text-white py-2 px-4 rounded-md hover:bg-blue-700 transition-colors duration-200">
        Book Now
      </button>
    </div>
  )
}

export default ServiceCard;