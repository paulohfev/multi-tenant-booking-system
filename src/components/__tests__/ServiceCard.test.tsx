import { render } from '@testing-library/react'
import ServiceCard from '../ServiceCard'

const mockService = {
  id: '1',
  centerId: '1',
  name: 'Test Service',
  description: 'Test Description',
  duration: 60,
  price: 100,
}

describe('ServiceCard', () => {
  it('should match snapshot', () => {
    const { container } = render(<ServiceCard service={mockService} />)
    expect(container).toMatchSnapshot()
  })
})
