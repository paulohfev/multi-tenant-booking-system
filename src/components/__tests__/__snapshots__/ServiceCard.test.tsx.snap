// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceCard should match snapshot 1`] = `
<div>
  <div
    class="bg-white rounded-lg shadow-md p-6 hover:shadow-lg transition-shadow duration-200"
  >
    <h3
      class="text-xl font-semibold text-gray-900 mb-2"
    >
      Test Service
    </h3>
    <p
      class="text-gray-600 mb-4"
    >
      Test Description
    </p>
    <div
      class="flex items-center justify-between text-sm text-gray-500 mb-4"
    >
      <span>
        Duration: 
        60
         min
      </span>
      <span
        class="font-semibold text-gray-900"
      >
        $
        100
      </span>
    </div>
    <button
      class="cursor-pointer w-full bg-blue-600 text-white py-2 px-4 rounded-md hover:bg-blue-700 transition-colors duration-200"
    >
      Book Now
    </button>
  </div>
</div>
`;
